server/src/main/java/org/elasticsearch/search/aggregations/package-info.java

```diff
@@ -7,6 +7,63 @@
  */
 
 /**
- * Builds analytic information over all hits in a search request.
+ * <h2>Aggregations</h2>
+ * <p>Builds analytic information over all hits in a search request.  Aggregations
+ * are essentially a tool for sumarizing data, and that summary is often used
+ * to generate a visualization.</p>
+ *
+ * <h2>Types of aggregations</h2>
+ * There are three main types of aggregations, each in their own sub package:
+ * <ul>
+ *     <li>Bucket aggregations - which group documents (e.g. a histogram)</li>
+ *     <li>Metric aggregations - which compute a summary value from several
+ *     documents (e.g. a sum)</li>
+ *     <li>Pipeline aggregations - which run as a seperate step and compute
+ *     values across buckets</li>
+ * </ul>
+ * Additionally there is a support sub package, which contains the type checking
+ * and resolution logic, primarily.
+ *
+ * <h2>How Aggregations Work</h2>
+ * <p>TODO: Info about search phases goes here</p>
+ *
+ * <p>Aggregations operate in general as Map Reduce jobs.  The coordinating node for
+ * the query dispatches the aggregation to each data node.  The data nodes all
+ * instantiate an {@link org.elasticsearch.search.aggregations.AggregationBuilder}
+ * of the appropriate type, which in turn builds the
+ * {@link org.elasticsearch.search.aggregations.Aggregator} for that node.  This
+ * collects the data from that shard, via
+ * {@link org.elasticsearch.search.aggregations.Aggregator#getLeafCollector(org.apache.lucene.index.LeafReaderContext)}
+ * more or less.  These values are shipped back to the coordinating node, which
+ * performs the reduction on them (partial reductions in place on the data nodes
+ * are also possible).</p>
+ *
+ * <h3>Three modes of operation</h3>
+ * <p>When it comes to actually collecting values, there are three ways aggregations
+ * operate, in general.  Which one we choose depends on limitations in the query
+ * and how the data was ingested (e.g. if it is searchable).</p>
+ *
+ * <p>The easiest to understand is the <strong>Compatible</strong> (TODO: Why is this called "compatible"?)
```

Comment:
Its the "lowest common denominator". There aren't any times when you can't use it.